[Flaky CI] compile error because of missing stdint.h

Example CI run: https://gitlab.tiker.net/inducer/loopy/-/jobs/235172

Continued from [here](https://github.com/inducer/loopy/pull/179#issuecomment-786174617):
```
_____________________________ test_c_optimizations _____________________________
[gw0] linux -- Python 3.9.1 /var/lib/gitlab-runner/builds/zCL2egrE/2/inducer/loopy/.env/bin/python3
Traceback (most recent call last):
  File "/var/lib/gitlab-runner/builds/zCL2egrE/2/inducer/loopy/.env/lib/python3.9/site-packages/pytools/__init__.py", line 682, in wrapper
    return getattr(obj, cache_dict_name)[key]
AttributeError: 'CKernelExecutor' object has no attribute '_memoize_dic_kernel_info'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/var/lib/gitlab-runner/builds/zCL2egrE/2/inducer/loopy/test/test_c_execution.py", line 169, in test_c_optimizations
    assert np.allclose(knl(a=a_np)[1], 2 * a_np)
  File "/var/lib/gitlab-runner/builds/zCL2egrE/2/inducer/loopy/loopy/kernel/__init__.py", line 1464, in __call__
    return kex(*args, **kwargs)
  File "/var/lib/gitlab-runner/builds/zCL2egrE/2/inducer/loopy/loopy/target/c/c_execution.py", line 452, in __call__
    kernel_info = self.kernel_info(self.arg_to_dtype_set(kwargs))
  File "/var/lib/gitlab-runner/builds/zCL2egrE/2/inducer/loopy/.env/lib/python3.9/site-packages/pytools/__init__.py", line 684, in wrapper
    result = function(obj, *args, **kwargs)
  File "/var/lib/gitlab-runner/builds/zCL2egrE/2/inducer/loopy/loopy/target/c/c_execution.py", line 427, in kernel_info
    c_kernels.append(CompiledCKernel(dp,
  File "/var/lib/gitlab-runner/builds/zCL2egrE/2/inducer/loopy/loopy/target/c/c_execution.py", line 345, in __init__
    self.dll = self.comp.build(self.name, self.code)
  File "/var/lib/gitlab-runner/builds/zCL2egrE/2/inducer/loopy/loopy/target/c/c_execution.py", line 271, in build
    compile_from_string(self.toolchain, name, code, c_fname,
  File "/var/lib/gitlab-runner/builds/zCL2egrE/2/inducer/loopy/.env/lib/python3.9/site-packages/codepy/jit.py", line 433, in compile_from_string
    toolchain.build_extension(ext_file, source_paths, debug=debug)
  File "/var/lib/gitlab-runner/builds/zCL2egrE/2/inducer/loopy/.env/lib/python3.9/site-packages/codepy/toolchain.py", line 210, in build_extension
    raise CompileError("module compilation failed")
codepy.CompileError: module compilation failed
----------------------------- Captured stderr call -----------------------------
/tmp/tmp_loopyv5p4j6wk/code.c: In function ‘loopy_kernel’:
/tmp/tmp_loopyv5p4j6wk/code.c:5:5: error: ‘int32_t’ undeclared (first use in this function)
    5 |     int32_t const k = 0;
      |     ^~~~~~~
/tmp/tmp_loopyv5p4j6wk/code.c:5:5: note: each undeclared identifier is reported only once for each function it appears in
/tmp/tmp_loopyv5p4j6wk/code.c:5:12: error: expected ‘;’ before ‘const’
    5 |     int32_t const k = 0;
      |            ^~~~~~
      |            ;
/tmp/tmp_loopyv5p4j6wk/code.c:8:14: error: expected ‘;’ before ‘const’
    8 |       int32_t const j = 0;
      |              ^~~~~~
      |              ;
/tmp/tmp_loopyv5p4j6wk/code.c:11:16: error: expected ‘;’ before ‘const’
   11 |         int32_t const i_outer = 0;
      |                ^~~~~~
      |                ;
FAILED compiler invocation:gcc -std=c99 -O3 -fPIC -shared -DNDEBUG -I/usr/include/python3.9 /tmp/tmp_loopyv5p4j6wk/code.c -L/usr/lib -lcrypt -lpthread -ldl -lutil -lm -o /tmp/tmp_loopyv5p4j6wk/0125eda4ad9b430080902b40832f06f8/loopy_kernel.so
```

cc @kaushikcfd

--- a/loopy/codegen/__init__.py
+++ b/loopy/codegen/__init__.py
@@ -524,6 +524,9 @@ def generate_code_v2(kernel):
         for idi in tv.decl_info(kernel.target, index_dtype=kernel.index_dtype):
             seen_dtypes.add(idi.dtype)
 
+    if kernel.all_inames():
+        seen_dtypes.add(kernel.index_dtype)
+
     preambles = kernel.preambles[:]
 
     preamble_info = PreambleInfo(

--- a/test/test_c_execution.py
+++ b/test/test_c_execution.py
@@ -351,6 +351,18 @@ def test_missing_compilers():
         __test(eval_tester, ExecutableCTarget, compiler=ccomp)
 
 
+def test_one_length_loop():
+    # https://github.com/inducer/loopy/issues/239
+    knl = lp.make_kernel(
+            "{[i]: 0<=i<1}",
+            """
+            a[i] = 42.0
+            """, target=lp.ExecutableCTarget())
+
+    _, (out, ) = knl()
+    assert out == 42
+
+
 if __name__ == "__main__":
     if len(sys.argv) > 1:
         exec(sys.argv[1])

